vector/src/androidTest/java/im/vector/app/ui/robot/NewRoomRobot.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2021 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.ui.robot
+
+import androidx.test.espresso.Espresso
+import com.adevinta.android.barista.interaction.BaristaClickInteractions
+import im.vector.app.R
+
+class NewRoomRobot {
+
+    fun createNewRoom(block: CreateNewRoomRobot.() -> Unit) {
+        BaristaClickInteractions.clickOn(R.string.create_new_room)
+        val createNewRoomRobot = CreateNewRoomRobot()
+        block(createNewRoomRobot)
+        if (!createNewRoomRobot.createdRoom) {
```

Comment:
the test should fail in this case, no?

Reply:
without this the next steps will fail as every robot assumes the start and end position are the same

in the case of creating a new room the room creation screen is automatically finished so there's no need to press back, if we did then we'd exit the app as the `RoomDetails` 


```kotlin
// move to room list tab and press #+
elementRobot.newRoom {
    // press create new room launches CreateRoomActivity
    createNewRoom {
        //  press create, navigates to the RoomDetailsActivity, CreateRoomActivity automatically closes
        createRoom {
            // goes through all the room detail checks
            crawl()             
            // on createRoom finish pressBack to take us back to root activity
        }
        // avoid pressing back because we created a room
    }
    // avoid pressing back because we created a room
}
```